### PR TITLE
Bugfix: fixes fuzzy messages arguments suggestion

### PR DESCRIPTION
**Pull request by lifeart**
fixes incorrect messaging in https://github.com/ember-template-lint/ember-template-lint/issues/2071

### CODE REVIEW

File: lib/rules/no-unknown-arguments-for-builtin-components.js [hunk 1]
```diff
@@ -292,28 +292,31 @@ function ERROR_MESSAGE(tagName, argumentName) {
...
     fuzzyResults = new Fuse(candidates).search(query);
     query = query.slice(0, -1);
   }
 
-  let closestKey = fuzzyResults.length ? `@${fuzzyResults[0].item}` : argumentName;
-
-  if (closestKey in deprecatedArgs) {
-    return deprecateArgument(tagName, closestKey, deprecatedArgs[closestKey]);
+  if (!hasArgumentsMatch) {
+    const msg = `"${argumentName}" is unknown argument for <${tagName} /> component.`;
```
Comment (scalvert) [inline lib/rules/no-unknown-arguments-for-builtin-components.js:303]: I would consider changing the message to this (and its associated test):

```suggestion
    const msg = `"${argumentName}" is not a known argument for the <${tagName} /> component.`;
```

File: test/unit/rules/no-unknown-arguments-for-builtin-components-test.js [hunk 2]
```diff
@@ -210,5 +210,21 @@ generateRuleTests({
...
+
+    // LinkTo unknown argument
+
+    {
+      template:
+        ' <LinkTo class="auk-search-results-list__item" @route={{@route}} @models={{this.models}} @random="test" @query={{@query}} ...attributes >Hello</LinkTo>',
+      results: [
+        {
+          message:
+            '"@random" is unknown argument for <LinkTo /> component. Did you mean "@dragOver"?',
```
Comment (rwjblue) [inline test/unit/rules/no-unknown-arguments-for-builtin-components-test.js:222]:
```suggestion
            '"@random" is not a known argument for <LinkTo /> component. Did you mean "@dragOver"?',
```

**Comment by lifeart** [comment]
I think we could introduce some refactoring helping rule here, like "require-valid-component-arguments", configurable by end-user (we could create els helper to discover it).

End-users could define / restrict arguments usage per component, having same API as in this rule.

```json
{
 'MyComponent': [ 'myArgument' ]
}
```

**Comment by lifeart** [comment]
node 16 ci failing because of V8 error:

image

**Review by scalvert — APPROVED** [review]

**Pull request merged**